### PR TITLE
[interp] Make the code GC safe on wasm.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by BrzVlad** [comment]
@vargaz Could we keep the volatility of `o` and indirection to `frame->o` just for TARGET_WASM ?

**Comment by vargaz** [comment]
Sure, but it would look a big ugly. Also, I don't think this change will hurt perf too much, its just one memory load from the stack.

**Comment by BrzVlad** [comment]
For the local obj maybe we can define something like local_obj to be either o or frame->o, while for the volatility is straight forward. Shouldn't impact the readability too much.

**Comment by vargaz** [comment]
Long term, we should centralize the handling of objrefs which would make it possible to do precise stack scanning in the interpreter, its much easier to do than in the jit.

File: mono/mini/interp/interp-internals.h [hunk 1]
```diff
@@ -137,7 +137,11 @@ struct _InterpFrame {
 	 * Storing into this field will keep the object pinned
 	 * until the objref can be stored into stackval->data.o.
 	 */
+#ifdef TARGET_WASM
 	MonoObject* volatile o;
+#else
```
Comment (BrzVlad) [inline mono/mini/interp/interp-internals.h]: This should go also for `stackval` structure. Well I guess only there, since `o` is dead here.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by lewing** [comment]
@monojenkins build Linux x64 C++

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**